mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxViewModelProvider.kt

```diff
@@ -22,68 +26,81 @@ object MvRxViewModelProvider {
      *                         Either [ActivityViewModelContext] or [FragmentViewModelContext].
      * @param key An optional key for the ViewModel in the store. This is optional but should be used if you have multiple of the same
      *            ViewModel class in the same scope.
+     * @param forExistingViewModel If true the viewModel should already have been created. If it has not been created already,
+     *                             a [ViewModelDoesNotExistException] will be thrown
+     * @param initialStateFactory A way to specify how to create the initial state, can be mocked out for testing.
+     *
      */
     fun <VM : BaseMvRxViewModel<S>, S : MvRxState> get(
         viewModelClass: Class<out VM>,
         stateClass: Class<out S>,
         viewModelContext: ViewModelContext,
         key: String = viewModelClass.name,
+        forExistingViewModel: Boolean = false,
         initialStateFactory: MvRxStateFactory<VM, S> = RealMvRxStateFactory()
     ): VM {
-        // This wraps the fact that ViewModelProvider.of has individual methods for Fragment and FragmentActivity.
-        val factory = MvRxFactory { createViewModel(viewModelClass, stateClass, viewModelContext, initialStateFactory = initialStateFactory) }
-        return when (viewModelContext) {
-            is ActivityViewModelContext -> ViewModelProviders.of(viewModelContext.activity, factory)
-            is FragmentViewModelContext -> ViewModelProviders.of(viewModelContext.fragment, factory)
-        }.get(key, viewModelClass)
-    }
+        val stateRestorer = viewModelContext
+            .savedStateRegistry
```

Comment:
did you address the case where this is called before onCreate and savedStateRegistry doesn't have the restored state yet?

Reply:
I can check if `savedStateRegistroy.isRestored` and if not throw a custom error message (it currently already throws an exception but the error message makes a little less sense to users who don't know about the saved state registry). I also added a test for this case.